BridgeAppSDK/ORKResult+SBAExtension.swift

```diff
@@ -0,0 +1,329 @@
+//
+
+import ResearchKit
+
+private var kStartDateKey = "startDate"
+private var kEndDateKey = "endDate"
+private var kIdentifierKey = "identifier"
+
+private var kItemKey = "item"
+
+private var kTappingViewSizeKey = "TappingViewSize"
+private var kButtonRectLeftKey = "ButtonRectLeft"
+private var kButtonRectRightKey = "ButtonRectRight"
+private var kTapTimeStampKey = "TapTimeStamp"
+private var kTapCoordinateKey = "TapCoordinate"
+private var kTappedButtonIdKey = "TappedButtonId"
+private var kTappedButtonNoneKey = "TappedButtonNone"
+private var kTappedButtonLeftKey = "TappedButtonLeft"
+private var kTappedButtonRightKey = "TappedButtonRight"
+private var kTappingSamplesKey = "TappingSamples"
+
+private var kSpatialSpanMemoryGameStatusKey = "MemoryGameStatus"
+private var kSpatialSpanMemoryGameStatusUnknownKey = "MemoryGameStatusUnknown"
+private var kSpatialSpanMemoryGameStatusSuccessKey = "MemoryGameStatusSuccess"
+private var kSpatialSpanMemoryGameStatusFailureKey = "MemoryGameStatusFailure"
+private var kSpatialSpanMemoryGameStatusTimeoutKey = "MemoryGameStatusTimeout"
+
+private var kSpatialSpanMemorySummaryNumberOfGamesKey = "MemoryGameNumberOfGames"
+private var kSpatialSpanMemorySummaryNumberOfFailuresKey = "MemoryGameNumberOfFailures"
+private var kSpatialSpanMemorySummaryOverallScoreKey = "MemoryGameOverallScore"
+private var kSpatialSpanMemorySummaryGameRecordsKey = "MemoryGameGameRecords"
+
+private var kSpatialSpanMemoryGameRecordSeedKey = "MemoryGameRecordSeed"
+private var kSpatialSpanMemoryGameRecordGameSizeKey = "MemoryGameRecordGameSize"
+private var kSpatialSpanMemoryGameRecordGameScoreKey = "MemoryGameRecordGameScore"
+private var kSpatialSpanMemoryGameRecordSequenceKey = "MemoryGameRecordSequence"
+private var kSpatialSpanMemoryGameRecordTouchSamplesKey = "MemoryGameRecordTouchSamples"
+private var kSpatialSpanMemoryGameRecordTargetRectsKey = "MemoryGameRecordTargetRects"
+
+private var kSpatialSpanMemoryTouchSampleTimeStampKey = "MemoryGameTouchSampleTimestamp"
+private var kSpatialSpanMemoryTouchSampleTargetIndexKey = "MemoryGameTouchSampleTargetIndex"
+private var kSpatialSpanMemoryTouchSampleLocationKey = "MemoryGameTouchSampleLocation"
+private var kSpatialSpanMemoryTouchSampleIsCorrectKey = "MemoryGameTouchSampleIsCorrect"
+
+private var QuestionResultQuestionTextKey = "questionText"
+private var QuestionResultQuestionTypeKey = "questionType"
+private var QuestionResultUserInfoKey = "userInfo"
+
+private var SurveyAnswerKey = "answer"
+
+class ArchiveableResult : NSObject {
+    let result: AnyObject
+    let filename: String
+    
+    init(result: AnyObject, filename: String) {
+        self.result = result
+        self.filename = filename
+        super.init()
+    }
+}
+
+protocol BridgeUploadableData {
+    // returns result object, result type, and filename
+    func bridgeData(stepIdentifier: String) -> ArchiveableResult?
+}
+
+extension ORKResult: BridgeUploadableData {
```

Comment:
Should make this public so that apps that have their own implementation of an ORKResult subclass can be archived.